support-frontend/app/controllers/CreateSubscriptionController.scala

```diff
@@ -46,8 +46,8 @@ class CreateSubscriptionController(
 
   type ApiResponseOrError[RES] = EitherT[Future, CreateSubscriptionError, RES]
 
-  def create: Action[CreateSupportWorkersRequest] =
```

Comment:
It's a shame to have lost the type info here, `EssentialAction` doesn't tell us much.
Do you think there's any value in separating it from the alarmOnFailure wrapping? E.g.
```
private def create: Action[CreateSupportWorkersRequest] = ...
def createWithAlarm = alarmOnFailure(create)
```

Reply:
thanks, see what you mean. Actually from a type checking point of view I don't think the type matters at this point in terms of outward facing, other than as documentation.   This is because the JSON parsing happens "inside" the `Action`. What matters is that the `request` on line 51 is the right type, which would be checked elsewhere.
I'll have a think about what makes sense here though, as I agree it looks strange to not have that.

Reply:
so I had a look and thought a bit more and still don't see a benefit from the type info
however when I created the existing LoggingCirceParser I didn't really understand Actions and BodyParsers, now I've done a bit more, we could probably change 
```scala
alarmOnFailure(maybeAuthenticatedAction().async(new LoggingCirceParser(components).requestParser) {
```
to
```scala
alarmOnFailure(maybeAuthenticatedAction().async(loggingRequestParser[CreateSupportWorkersRequest]) {
```
or even
```scala
alarmOnFailureWithLogging(maybeAuthenticatedAction().async(circe.json[CreateSupportWorkersRequest]) {
```
but that last one might be another day or two of head scratching!